From ac3216dc39c443037978d7dea4c401565108e162 Mon Sep 17 00:00:00 2001
From: Gyorgy Sarvari <skandigraun@gmail.com>
Date: Mon, 12 Aug 2024 15:19:00 +0200
Subject: [PATCH] fix text input displaying

The virtual keyboard should be only enabled in case there is an *active, focused*
text input available. Currently, vanilla wlroots displays the keyboard even if there
is a non-focused text input somewhere in the workspace, which makes it quite
unusable.

This patch displays the keyboard only in case a text input is active.

Upstream-Status: Inappropriate
---
 types/wlr_text_input_v3.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/types/wlr_text_input_v3.c b/types/wlr_text_input_v3.c
index 0b05ca65..fe0dae96 100644
--- a/types/wlr_text_input_v3.c
+++ b/types/wlr_text_input_v3.c
@@ -181,7 +181,8 @@ static void text_input_commit(struct wl_client *client,
 	text_input->current_serial++;
 
 	if (text_input->focused_surface == NULL) {
-		wlr_log(WLR_DEBUG, "Text input commit received without focus");
+		wlr_log(WLR_ERROR, "Text input commit received without focus, returning");
+		return;
 	}
 
 	if (!old_enabled && text_input->current_enabled) {
-- 
2.45.2

